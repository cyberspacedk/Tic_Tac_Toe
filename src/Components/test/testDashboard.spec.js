import React from "react";
import { Provider } from "react-redux";
import { shallow, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import configureStore from "redux-mock-store";
import  DashboardConnected, {Dashboard} from "../Dashboard";

describe("Dashboard of game", () => {

  // 1- check. is component render with store props
  // 2- snapshot

  const mockClick = jest.fn();
  const mockStore = configureStore();
  let store, myComponent;

  const initialState = {
    history: [{squares:Array(9).fill("")}],
    turn: true,
    snapshot: 2,
    makeMove: mockClick,
    playerTurn: mockClick,
    switchSnap: mockClick
  };
 
  beforeEach(() => {
    store = mockStore(initialState);
    myComponent = shallow(<DashboardConnected store={store}/>);
  });

  it("Check is Dashboard render", () => { 
    expect(myComponent.length).toEqual(1);
  });

  it("Dashboard snapshot", () => {  
    expect(shallowToJson(myComponent)).toMatchSnapshot();
  });
  
});
