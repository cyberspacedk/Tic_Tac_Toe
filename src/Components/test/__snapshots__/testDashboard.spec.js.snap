// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard of game Dashboard snapshot 1`] = `
<ContextProvider
  value={null}
>
  <Dashboard
    history={
      Array [
        Object {
          "squares": Array [
            "",
            "",
            "",
            "",
            "",
            "",
            "",
            "",
            "",
          ],
        },
      ]
    }
    makeMove={[Function]}
    playerTurn={[Function]}
    snapshot={2}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    switchSnap={[Function]}
    turn={true}
  />
</ContextProvider>
`;
